### ISSUE TITLE
Warn about jQuery usage on the front end

### ISSUE THREAD

**Issue by adamsilverstein** [Otto42/theme-check#402]
While it has been widely known that [you might not need jQuery](https://youmightnotneedjquery.com/) for most of your interactive theme features, jQuery is still a common JavaScript dependency found in WordPress themes - for example, 7 out of the 11 default themes since Twenty Ten use jQuery in the frontend. 

A recent analysis of [the performance impact of using jQuery in WordPress themes](https://make.wordpress.org/themes/2021/10/04/the-performance-impact-of-using-jquery-in-wordpress-themes/
) determined that **jQuery is the most common JavaScript-based performance problem in themes** and that **Removing jQuery may lead to 80% less JavaScript load**. According to the analysis, most themes can easily substitute (modern) vanilla JavaScript and remove their reliance on jQuery with only minor changes.

To help theme developers make better choices in the future, I would like to add a warning to the theme-check plugin when themes enqueue 'jquery' on the front end (eg. `wp_enqueue_script`). The warning could reference the analysis and perhaps offer a link to an [article explaining how to switch to vanilla JavaScript](https://tobiasahlin.com/blog/move-from-jquery-to-vanilla-javascript/). 

If this idea seems reasonable, I am happy to work on a PR to add this.

**Comment by carolinan** [comment]
Yes,  I had it on my to do list but if you want to work on it that would be great.

There is also the scenario where the user tries to include and enqueue their own version of jQuery -that should block theme upload.

**Comment by adamsilverstein** [comment]
@carolinan finally found time to get back to this and opened https://github.com/WordPress/theme-check/pull/423 to address.

I tested this locally to verify it works :) Still have some questions about the implementation which I will leave as comments on the PR code
